### PR TITLE
Handle ECS race condition for non-existent service

### PR DESCRIPTION
**Pull request by mikhail**
@swaggy @niallo 

@diranged

### CODE REVIEW

**Comment by swaggy** [comment]
LGTM

File: kingpin/actors/aws/ecs.py [hunk 1]
```diff
@@ -753,13 +756,12 @@ def _describe_service(self, service_name):
         services = response['services']
 
         # There should never be more than one service for a given name.
-        if len(services) > 1:
+        if len(services) != 1:
```
Comment (diranged) [inline kingpin/actors/aws/ecs.py]: Wait .. this means that if there is no service, it raises an exception right? Doesn't that mean that we can't create a service if its missing?

Comment (swaggy) [inline kingpin/actors/aws/ecs.py]: Er, yeah, that's right. We need to catch that exception where it's called.

Comment (diranged) [inline kingpin/actors/aws/ecs.py]: In that case, it shouldnt be a `RecoverableActorFailure` ... it should be a custom exception and let the caller catch that exception and handle it.

**Pull request merged**